Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java

```diff
@@ -0,0 +1,261 @@
+package nl.soccar.gameserver.model.session;
+
+import nl.soccar.gameserver.GameServerConstants;
+import nl.soccar.gameserver.controller.rmi.GameServerRmiController;
+import nl.soccar.gameserver.controller.socnet.message.*;
+import nl.soccar.gameserver.model.GameServer;
+import nl.soccar.gameserver.model.PlayerWrapper;
+import nl.soccar.gameserver.model.command.Command;
+import nl.soccar.gameserver.model.command.CommandDispatcher;
+import nl.soccar.gameserver.util.RoomUtilities;
+import nl.soccar.library.Player;
+import nl.soccar.library.Room;
+import nl.soccar.library.Team;
+import nl.soccar.library.enumeration.Privilege;
+import nl.soccar.library.enumeration.TeamColour;
+import nl.soccar.socnet.connection.Connection;
+
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * @author Lesley
+ */
+public final class RoomWrapper {
+
+    private static final Logger LOGGER = Logger.getLogger(RoomWrapper.class.getSimpleName());
+    private static final Random RANDOM = new Random();
+
+    private final List<PlayerWrapper> players = new ArrayList<>();
+    private final Map<PlayerWrapper, Team> teams = new HashMap<>();
+
+    private final List<String> mutedNames = new ArrayList<>();
+
+    private final SessionWrapper session;
+    private final Room room;
+
+    public RoomWrapper(SessionWrapper session, Room room) {
+        this.session = session;
+        this.room = room;
+    }
+
+    public void join(PlayerWrapper player, String password) {
+        if (room.getOccupancy() + 1 > room.getCapacity()) {
+            player.getConnection().send(new JoinSessionMessage(JoinSessionMessage.Status.CAPACITY_OVERFLOW));
+            return;
+        }
+
+        if (!room.check(password)) {
+            player.getConnection().send(new JoinSessionMessage(JoinSessionMessage.Status.INVALID_PASSWORD));
+            return;
+        }
+
+        Player unwrapped = player.unwrap();
+        synchronized (players) {
+            if (players.stream()
+                    .map(PlayerWrapper::unwrap)
+                    .anyMatch(p -> p.equals(unwrapped))) {
+                player.getConnection().send(new JoinSessionMessage(JoinSessionMessage.Status.USERNAME_EXISTS));
+                return;
+            }
+
+            players.add(player);
+
+            unwrapped.setPlayerId(RoomUtilities.getNextPlayerId(this));
+            unwrapped.setCurrentSession(session.unwrap());
+        }
+
+        joinRandomTeam(player);
+        sendRoomInformation(player);
+
+        String roomName = room.getName();
+        LOGGER.log(Level.INFO, "Player {0} joined session {1}.", new String[]{player.getUsername(), roomName});
+
+        GameServerRmiController rmiController = GameServer.getInstance().getRmiControler();
+        rmiController.increaseOccupancy(roomName);
+        rmiController.changeHost(roomName, room.getHost().getUsername());
+    }
+
+    public void leave(PlayerWrapper player) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)